mlir/tools/miopen-gen/miopen-gen.cpp

```diff
@@ -1654,6 +1654,7 @@ static LogicalResult populateTensorFillLogic(mlir::OpBuilder &b,
 static LogicalResult
 populateHostHarnessLogic(ModuleOp &module,
                          const SmallVector<KernelIF, 8> &kernels,
+                         const SmallVector<KernelIF, 8> &roots,
```

Comment:
Call-graph roots are separate from kernels, though a function may be in both sets.